pyscriptjs/src/components/pyrepl.ts

```diff
@@ -217,9 +220,14 @@ export class PyRepl extends BaseEvalElement {
 
     getSourceFromElement(): string {
         const sourceStrings = [
-            `output_manager.change("` + this.outputElement.id + `")`,
+            `output_manager.change(out="${this.outputElement.id}", append=${this.appendOutput ? 'True' : 'False'})`,
             ...this.editor.state.doc.toString().split('\n'),
         ];
+
+        if(!this.appendOutput) {
+            sourceStrings.splice(2, 0, `output_manager.change(out="${this.outputElement.id}", append=True)`);
```

Comment:
This breaks in indented blocks.
Running:
```py
for i in range(10):
    print(i)
```
Results in a `JsException(PythonError: Traceback (most recent call last): File "/lib/python3.10/site-packages/_pyodide/_base.py", line 421, in eval_code CodeRunner( File "/lib/python3.10/site-packages/_pyodide/_base.py", line 237, in __init__ self.ast = next(self._gen) File "/lib/python3.10/site-packages/_pyodide/_base.py", line 141, in _parse_and_compile_gen mod = compile(source, filename, mode, flags | ast.PyCF_ONLY_AST) File "", line 3 output_manager.change(out="my-repl-1", append=True) ^^^^^^^^^^^^^^ IndentationError: expected an indented block after 'for' statement on line 2 )`

The reason for this is how you modify `sourceStrings`.
```typescript
const sourceStrings = [
            `output_manager.change(out="${this.outputElement.id}", append=${this.appendOutput ? 'True' : 'False'})`,
            ...this.editor.state.doc.toString().split('\n'),
];
const oldSourceStrings = [...sourceStrings];
console.log(oldSourceStrings);
if(!this.appendOutput) {
    sourceStrings.splice(2, 0, `output_manager.change(out="${this.outputElement.id}", append=True)`);
}
console.log(sourceStrings);
```
Yields:
```
0: "output_manager.change(out=\"my-repl-1\", append=False)"
1: "for i in range(10):"
2: "  print(i)"
```

and 
```
0: "output_manager.change(out=\"my-repl-1\", append=False)"
1: "for i in range(10):"
2: "output_manager.change(out=\"my-repl-1\", append=True)"
3: "  print(i)"
```

Line 2 in the 2nd output breaks the code.

Reply:
Good catch, I will think about a solution to that. Feel free to add a commit if you have a solution.
Thanks!

Reply:
I've spent an hour to find out why it breaks :smile:. I didn't think about a solution yet

Reply:
I added a commit with a relatively simple change. Should work now.
Instead of modifying the source, it resets the innerHTML of the outputElement, if output mode is set to replace.

Reply:
Thanks a lot for testing this @yannickfunk! 
I will resolve this, as the last commit should have fixed it.